financial-templates-lib/UniswapPriceFeed.js

```diff
@@ -5,15 +5,24 @@ const { MAX_SAFE_JS_INT } = require("../common/Constants");
 
 // A client for getting price information from a uniswap market.
 class UniswapPriceFeed {
-  constructor(abi, web3, uniswapAddress, twapLength, getTime) {
+  constructor(abi, web3, uniswapAddress, twapLength, historicalLookback, getTime) {
     this.web3 = web3;
     this.uniswap = new web3.eth.Contract(abi, uniswapAddress);
     this.twapLength = twapLength;
     this.getTime = getTime;
+    this.historicalLookback = historicalLookback;
   }
 
   getCurrentPrice = () => this.currentPrice;
   getCurrentTwap = () => this.currentTwap;
+  getHistoricalTwap = time => {
+    if (time < this.lastUpdateTime - this.historicalLookback) {
+      // Requesting an historical TWAP earlier than the lookback.
+      return null;
```

Comment:
I think you should throw an error here. As a caller I'd expect an answer or an error to be thrown.

Reply:
@chrismaree I agree with your above comment that returning null would be consistent, but as a user, do y'all think throwing is always preferable to returning null? For instance, if there are no Sync events in or before the window, we also return null. Should those be errors too?

Reply:
I agree it would be consistent with `getCurrentPrice()`. Thinking about this again, I think `null/undefined` is probably the best default value for `this.currentPrice` and therefore `getCurrentPrice()/getHistoricalPrice()` will just return `null` by default.

Most importantly is consistency, so looks good to me as is.

Reply:
Okay, SGTM. I'm honestly okay with either. I do think that the caller should only have to deal with one type of invalid response, though -- either they should wrap in a try-catch or they should check for null. I'll plan to leave as is for now, but if you do prefer to throw, I'm totally cool with changing all of them.